packages/rum-core/src/domain/rumEventsCollection/view/trackViews.spec.ts

```diff
@@ -208,6 +209,166 @@ describe('rum track url change', () => {
+    expect(getViewEvent(2).name).toBe('Bar from /foo')
+    expect(getViewEvent(4).name).toBeUndefined()
+    expect(getViewEvent(6).name).toBe('Foo from /baz')
+  })
+
+  it('should not create view for the same URL when shouldCreateView is undefined', () => {
+    onNewLocation = (location) => {
+      switch (location.pathname) {
+        case '/foo':
+          return { viewName: 'Foo' }
+        case '/bar':
+          return { viewName: 'Bar' }
+      }
+    }
+    setupBuilder.build()
+    history.pushState({}, '', '/foo')
+    history.pushState({}, '', '/bar')
+    history.pushState({}, '', '/bar')
+    history.pushState({}, '', '/foo')
+
+    expect(getViewEvent(0).name).toBe('Foo')
+    expect(getViewEvent(2).id).toBe(getViewEvent(0).id)
+    expect(getViewEvent(3).name).toBe('Bar')
+    expect(getViewEvent(5).id).toBe(getViewEvent(3).id)
+    expect(getViewEvent(6).name).toBe('Foo')
+  })
+
+  it('should ignore the view when shouldCreateView is false', () => {
+    onNewLocation = (location) => {
+      switch (location.pathname) {
+        case '/foo':
+          return { viewName: 'Foo', shouldCreateView: true }
+        case '/bar':
+          return { shouldCreateView: false }
+        case '/baz':
+          return { viewName: 'Baz', shouldCreateView: true }
+      }
+    }
+    setupBuilder.build()
+    history.pushState({}, '', '/bar')
+    history.pushState({}, '', '/baz')
+
+    const initialViewId = getViewEvent(0).id
+    expect(getViewEvent(0).name).toBe('Foo')
+    expect(getViewEvent(2).name).toBe('Foo')
+    expect(getViewEvent(2).id).toBe(initialViewId)
+    expect(getViewEvent(3).name).toBe('Baz')
+    expect(getViewEvent(3).id).not.toBe(initialViewId)
+  })
+
+  it('should create the initial view even when shouldCreateView is false', () => {
+    onNewLocation = (location) => {
+      if (location.pathname === '/foo') {
+        return { shouldCreateView: false }
+      }
+      if (location.pathname === '/bar') {
+        return { shouldCreateView: true }
+      }
+    }
+    setupBuilder.build()
+    history.pushState({}, '', '/bar')
+    history.pushState({}, '', '/foo')
+
+    expect(getViewEvent(0).location.pathname).toBe('/foo')
+    expect(getViewEvent(2).location.pathname).toBe('/bar')
+    expect(getViewEvent(4)).toBeUndefined()
+  })
+
+  it('should catch thrown errors', () => {
+    const fooError = 'Error on /foo path'
+    const barError = 'Error on /bar path'
+    onNewLocation = (location) => {
+      if (location.pathname === '/foo') {
+        throw fooError
+      }
+      if (location.pathname === '/bar') {
+        throw barError
+      }
+      return undefined
+    }
+    const consoleErrorSpy = spyOn(console, 'error')
+    setupBuilder.build()
+    expect(consoleErrorSpy).toHaveBeenCalledWith('onNewLocation threw an error:', fooError)
+    history.pushState({}, '', '/bar')
+    expect(consoleErrorSpy).toHaveBeenCalledWith('onNewLocation threw an error:', barError)
```

Comment:
what is the reasoning for having two error cases?

Reply:
The first one happens on the initial view; the second on location change.